sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/PartitionPublishingProperties.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Messaging.EventHubs.Producer
+{
+    /// <summary>
+    ///   A set of information for an Event Hub.
+    /// </summary>
+    ///
+    public class PartitionPublishingProperties
+    {
+        /// <summary>
+        ///   Indicates whether or not idempotent publishing is enabled for the producer and, by extension, the associated partition.
+        /// </summary>
+        ///
+        /// <value><c>true</c> if the idempotent publishing is enabled; otherwise, <c>false</c>.</value>
+        ///
+        public bool IsIdempotentPublishingEnabled { get; }
+
+        /// <summary>
+        ///   The identifier of the producer group for which this producer is publishing to the associated partition.
+        /// </summary>
+        ///
+        public long? ProducerGroupId  { get; set; }
+
+        /// <summary>
+        ///   The owner level of this producer for publishing to the associated partition.
+        /// </summary>
+        ///
+        public short? OwnerLevel { get; set; }
+
+        /// <summary>
+        ///   The sequence number assigned to the event that was most recently published to the associated partition
+        ///   successfully.
+        /// </summary>
+        ///
+        public int? LastPublishedSequenceNumber   { get; set; }
```

Comment:
We kind of discussed this before - should we come up with a different name for this to avoid confusion/overlap with the SequenceNumber that most people are going to interact with?

Would it be bad to call it a Count, for instance? Since it has to be ordinal and can't contain gaps?
(just spitballing)

Reply:
Naming is still in the "here's what the working group accepted during design reviews" stage.   I absolutely expect changes as we move into arch board reviews and such.

That said "count" here wouldn't be accurate, I don't think.  I don't believe that it adequately communicates the scope "count sent for this particular producer group and owner level combination which may have started at an arbitrary point."    While "sequence number" also lacks that level of insight, I think that it doesn't come with a strong a connotation that it starts at 0|1 and counts up.

Reply:
Yes, naming is always the hardest part. 

Is the idea of having this be a "hidden" property still on the table? Or are we dead-set on exposing the value?